webcrypto/hmac.go

```diff
@@ -100,13 +100,13 @@ func (hkgp *HMACKeyGenParams) GenerateKey(
 	if !hkgp.Length.Valid {
 		switch hkgp.Hash.Name {
 		case SHA1:
-			hkgp.Length = null.IntFrom(sha1.BlockSize * 8)
+			hkgp.Length = null.IntFrom(int64(byteLength(sha1.BlockSize).asBitLength()))
 		case SHA256:
-			hkgp.Length = null.IntFrom(sha256.BlockSize * 8)
+			hkgp.Length = null.IntFrom(int64(byteLength(sha256.BlockSize).asBitLength()))
 		case SHA384:
-			hkgp.Length = null.IntFrom(sha512.BlockSize * 8)
+			hkgp.Length = null.IntFrom(int64(byteLength(sha512.BlockSize).asBitLength()))
 		case SHA512:
-			hkgp.Length = null.IntFrom(sha512.BlockSize * 8)
+			hkgp.Length = null.IntFrom(int64(byteLength(sha512.BlockSize).asBitLength()))
```

Comment:
This will be *slightly* easier if we had `nullBitLength` but I don't know if it is worth doing in practice.

I would though recommend having a variable that is set in the cases and moving the actual `null.IntFrom` after the select.

Reply:
Funnily enough, I had started implementing `nullBitLength`, but quickly reconsidered as it was a lot of code introduced for a minimal use case at this point. We might reconsider at a later point in time if this pattern reemerges :+1: